sw/device/lib/runtime/ibex.h

```diff
@@ -152,6 +152,17 @@ inline bool ibex_timeout_check(const ibex_timeout_t *timeout) {
   return ibex_mcycle_read() - timeout->start < timeout->cycles;
 }
 
+/**
+ * Returns the time elapsed in microseconds since the `ibex_timeout_init` was
+ * called.
+ *
+ * @param timeout The `ibex_timeout_t` value.
+ * @return Time elapsed in microseconds.
+ */
+inline uint32_t ibex_timeout_elapsed(const ibex_timeout_t *timeout) {
```

Comment:
I think this should return a `uint64_t` since otherwise the elapsed time will wrap at ~1 hour.

Reply:
That was my inicial plan, but the funcion uses `ibex_mcycle_read` which returns a `uint32_t` because the `mepc` is a 32 bit register.

Reply:
`ibex_mcycle_read()` returns a `uint64_t`, see [line 45 above](https://github.com/lowRISC/opentitan/blob/616eadc1a1133ac45421f32ecc8d4458ca1daf4c/sw/device/lib/runtime/ibex.h#L45). Am I missing something here?

Reply:
You are right, I was looking at the function `ibex_mepc_read`. I'll change it to `uint32_t` thanks